### PR TITLE
MAUI-3889 UG documentation for Rating control

### PR DESCRIPTION
**Pull request by EwaldJebaraj**
UG Added for .NET MAUI Rating control

### CODE REVIEW

File: MAUI/Rating/Appearance-Customization.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+description: Learn here all about Appearance Customization support in Syncfusion .NET MAUI Rating (SfRating) control and more.
+platform: maui
+control: Rating
+documentation: ug
+
+---
+
+# Appearance Customization in .NET MAUI Rating (SfRating)
+
+We can customize the rated and unrated fill color, stroke and stroke thickness using the following properties of [`RatingSettings`].
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Appearance-Customization.md:14]: You can customize the rated and unrated fill color, stroke, and stroke thickness using the following properties of [`RatingSettings`].

File: MAUI/Rating/Appearance-Customization.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+* [`RatedStrokeThickness`]
+* [`UnratedStrokeThickness`]
+ 
+## Set Fill Color
+
+[`SfRating`] control has support to set the fill color for the rated and unrated items.
+
+### Rated Items
+
+The [`RatedFill`] property fills the rated item with the specified solid color in the [`SfRating`]
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Appearance-Customization.md:29]: Please keep full stop at the end

File: MAUI/Rating/Appearance-Customization.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+* [`RatedStroke`]
+* [`UnratedStroke`]
+* [`RatedStrokeThickness`]
+* [`UnratedStrokeThickness`]
+ 
+## Set Fill Color
+
+[`SfRating`] control has support to set the fill color for the rated and unrated items.
+
+### Rated Items
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Appearance-Customization.md:27]: Rated items

File: MAUI/Rating/Appearance-Customization.md (hunk 2 above)
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Appearance-Customization.md:29]: The [`RatedFill`] property fills the rated items with the specified solid color in the [`SfRating`].

File: MAUI/Rating/Appearance-Customization.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+           
+{% endhighlight %}
+
+{% endtabs %}
+
+![Rated item fill color](images/rated-fill.png)
+
+### Unrated Items
+
+The [`UnratedFill`] property fills the unrated item with the specified solid color in the [`SfRating`].
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Appearance-Customization.md:59]: The [`UnratedFill`] property fills the unrated items with the specified solid color in the [`SfRating`].

File: MAUI/Rating/Appearance-Customization.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+{% endtabs %}
+
+![Unrated item fill color](images/unrated-fill.png)
+
+## Set Stroke
+
+[`SfRating`] control has support to set the stroke color for the rated and unrated items.
+
+### Rated Items
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Appearance-Customization.md:91]: Rated items

File: MAUI/Rating/Appearance-Customization.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+The [`RatedStroke`] property sets the stroke for the rated item with the specified solid color for the rated items in the [`SfRating`] control.
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Appearance-Customization.md:93]: The [`RatedStroke`] property sets the stroke for the rated items with the specified solid color in the [`SfRating`] control.

File: MAUI/Rating/Appearance-Customization.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+{% endtabs %}
+
+![Unrated item stroke](images/unrated-stroke.png)
+ 
+## Set Stroke Thickness
+
+[`SfRating`] control has support to set the stroke thickness for the rated and unrated items.
+
+### Rated Items
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Appearance-Customization.md:155]: Rated items

File: MAUI/Rating/Appearance-Customization.md [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+
+The [`RatedStrokeThickness`] property sets the stroke thickness for the rated item with the specified value in the [`SfRating`] control.
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Appearance-Customization.md:157]: The [`RatedStrokeThickness`] property sets the stroke thickness for the rated items with the specified value in the [`SfRating`] control.

File: MAUI/Rating/Getting-Started.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+platform: maui
+control: Rating
+documentation: ug
+
+---
+
+# Getting Started with .NET MAUI Rating (SfRating)
+
+
+This section explains how to configure a [`SfRating`] control in a real-time scenario and also provides a walk-through on some of the customization features available in [`SfRating`] control.
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Getting-Started.md:15]: This section explains how to configure a [`SfRating`] control in a real-time scenario and also provides a walk-through on some of the customization features available in [`SfRating`] control.

File: MAUI/Rating/Getting-Started.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+    }
+}   
+
+{% endhighlight %} 
+
+## Adding the .NET MAUI Rating control
+
+Step 1: Add the NuGet to the project as discussed in the above reference section. 
+
+Step 2: Add the namespace as shown in the following code sample:
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Getting-Started.md:61]: Step 2: Add the namespace, as shown in the following code sample.

File: MAUI/Rating/Getting-Started.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+    this.Content = rating;
+} 
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Set Number of Rating Items
+
+The number of rating items to be displayed can be customized in the [`SfRating`] control. Users can create a rating application with 5 items as follows. The [`ItemCount`] property is used to define the number of rating items.
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Getting-Started.md:107]: Users can create a rating application using five items as follows.

File: MAUI/Rating/Getting-Started.md [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+    rating.ItemCount = 5;
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Set Value
+
+The display value can be set in the [`SfRating`] control, which is selected among the items. The following code example shows the display value of 3 with 5 rating items. The [`Value`] property is used to set display value.
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Getting-Started.md:135]: The display value can be set in the [`SfRating`] control, which is selected among the items. The following code example shows the display value of three with five rating items. The [`Value`] property is used to set the display value.

File: MAUI/Rating/Getting-Started.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+    rating.Value = 3;
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Precision
+
+The[`SfRating`] control provides an option to rate the items in full, half, and exact values. This can be set using the [`Precision`] property. By default, the precision mode is [`Standard`].
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Getting-Started.md:163]: The [`SfRating`] control provides an option to rate the items in full, half, and exact values.

File: MAUI/Rating/Overview.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+title: About .NET MAUI Rating control | Syncfusion
+description: Learn here all about the introduction of Syncfusion .NET MAUI Rating (SfRating) control, its elements and more.
+platform: maui
+control: Rating
+documentation: ug
+
+---
+# .NET MAUI Rating (SfRating) Overview
+
+The Essential .NET MAUI Rating control provides the number of stars that represents a rating. And also used to configure the item size, item spacing and the number of displayed items in the [`SfRating`] control. Essential .NET MAUI Rating control can be used in various scenarios like movie rating, rating the application etc.
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Overview.md:13]: The Essential .NET MAUI Rating control provides the number of stars that represents a rating. Also used to configure the item size, item spacing, and the number of displayed items in the [`SfRating`] control. Essential .NET MAUI Rating control can be used in various scenarios like movie rating, rating the application, etc.

File: MAUI/Rating/Overview.md [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+![Rating OverView](images/overview.png)
+
+## Key Features
+
+* [`Precision`]: Options to decide the accuracy level of rating.
+
+* [`ItemCount`]: Support to determine the number of Rating items to be displayed.
+
+* ['RatingShape']: Predefined shapes for the rating item
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Overview.md:23]: keep full stop at the end.

File: MAUI/Rating/Rating-Shape.md [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+    rating.RatingShape = RatingShape.Custom;
+    rating.Path = "M17.5 35.5C19.9063 35.5 21.875 33.8846 21.875 31.9103H13.125C13.125 33.8846 15.0719 35.5 17.5 35.5ZM30.625 24.7308V15.7564C30.625 10.2462 27.0375 5.63334 20.7812 4.41282V3.19231C20.7812 1.70256 19.3156 0.5 17.5 0.5C15.6844 0.5 14.2188 1.70256 14.2188 3.19231V4.41282C7.94063 5.63334 4.375 10.2282 4.375 15.7564V24.7308L0 28.3205V30.1154H35V28.3205L30.625 24.7308Z";
+
+}
+
+{% endhighlight %} 
+
+{% endtabs %}
+
+N> The Item size should be same as the path size for Custom Rating Shape
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Rating-Shape.md:56]: The Item size should be the same as the path size for Custom Rating Shape.

File: MAUI/Rating/Restrict-User-Selection.md [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+description: Learn here all about Restrict User Selection support in Syncfusion .NET MAUI Rating (SfRating) control and more.
+platform: maui
+control: Rating
+documentation: ug
+
+---
+
+# Restrict user selection in .NET MAUI Rating (SfRating)
+
+[`SfRating`] control provides support for changeable or unchangeable values for Rating control. This is achieved by the [`IsReadOnly`] property. When this property is set to True, the Rating value becomes unchangeable. By default, this property value is set to False.
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Restrict-User-Selection.md:14]: [`SfRating`] control supports changeable or unchangeable values for Rating control.

File: MAUI/Rating/Migration.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+title: Migrating from Xamarin to .NET MAUI SfRating | Syncfusion 
+description: Learn all about Migrating from Syncfusion Xamarin Rating to Syncfusion .NET MAUI Rating control and more here.
+platform: maui
+control: SfRating
+documentation: ug
+---  
+
+# Migrating from Xamarin SfRating to .NET MAUI SfRating 
+
+To migrate easier from Xamarin SfRating to [.NET MAUI SfRating], we kept most of the APIs from Xamarin SfRating in MAUI SfRating. However, to maintain the consistency of API naming in MAUI SfRating, we renamed some of the APIs. The APIs changed in MAUI SfRating from Xamarin SfRating are detailed as follows:
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Migration.md:12]: To migrate easier from Xamarin SfRating to [.NET MAUI SfRating], most of the APIs were kept from Xamarin SfRating in MAUI SfRating. However, to maintain the consistency of API naming in MAUI SfRating, some of the APIs were renamed. The APIs changed in MAUI SfRating from Xamarin SfRating are detailed as follows:

File: MAUI/Rating/Getting-Started.md [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+    rating.Precision = Precision.Standard;
+}
+
+{% endhighlight%}
+
+{% endtabs %}
+
+![SfRating Getting Started ](images/getting-started.png)
+
+The complete Getting Started sample is available in this [link].
```
Comment (laksmipriyamurugesan) [inline MAUI/Rating/Getting-Started.md:189]: The complete Getting Started sample is available in this [link]

Add a hyperlink to the `link`.

Comment (EwaldJebaraj) [reply MAUI/Rating/Getting-Started.md:189]: It will be added once released

File: MAUI/Rating/Migration.md (hunk 18 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Migration.md]: Changed

File: MAUI/Rating/Appearance-Customization.md (hunk 1 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md]: Changed

File: MAUI/Rating/Appearance-Customization.md (hunk 3 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md]: Changed

File: MAUI/Rating/Appearance-Customization.md (hunk 2 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md]: Changed

File: MAUI/Rating/Appearance-Customization.md (hunk 4 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md]: Changed

File: MAUI/Rating/Appearance-Customization.md [hunk 20]
```diff
@@ -0,0 +1,213 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Unrated item fill color](images/unrated-fill.png)
+
+## Set Stroke
+
+[`SfRating`] control has support to set the stroke color for the rated and unrated items.
```
[the remark this answers is not in the document]
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md:89]: No changes

File: MAUI/Rating/Appearance-Customization.md (hunk 5 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md]: Changed

File: MAUI/Rating/Appearance-Customization.md (hunk 6 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md]: Changed

File: MAUI/Rating/Appearance-Customization.md (hunk 7 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md]: Changed

File: MAUI/Rating/Appearance-Customization.md (hunk 8 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md]: Changed

File: MAUI/Rating/Appearance-Customization.md [hunk 21]
```diff
@@ -0,0 +1,213 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Rated item stroke thickness](images/rated-stroke-thickness.png)
+
+### Unrated Items
+
+The [`UnratedStrokeThickness`] property sets the stroke thickness for the unrated item with the specified value in the [`SfRating`] control.
```
[the remark this answers is not in the document]
Comment (EwaldJebaraj) [reply MAUI/Rating/Appearance-Customization.md]: Changed

File: MAUI/Rating/Getting-Started.md (hunk 9 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Getting-Started.md:15]: No Changes

File: MAUI/Rating/Getting-Started.md (hunk 10 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Getting-Started.md]: Changed

File: MAUI/Rating/Getting-Started.md (hunk 11 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Getting-Started.md:107]: Changed

File: MAUI/Rating/Getting-Started.md (hunk 12 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Getting-Started.md]: Changed

File: MAUI/Rating/Getting-Started.md (hunk 13 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Getting-Started.md]: Changed

File: MAUI/Rating/Overview.md (hunk 14 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Overview.md]: Changed

File: MAUI/Rating/Overview.md (hunk 15 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Overview.md]: Changed

File: MAUI/Rating/Rating-Shape.md (hunk 16 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Rating-Shape.md]: Changed

File: MAUI/Rating/Restrict-User-Selection.md (hunk 17 above)
Comment (EwaldJebaraj) [reply MAUI/Rating/Restrict-User-Selection.md]: Changed

**Review by PaulAndersonS — APPROVED** [review]

**Pull request merged**